### PR TITLE
LOAD CSV parser provides input source description properly

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResources.scala [hunk 1]
```diff
@@ -39,7 +39,10 @@ class CSVResources(cleaner: TaskCloser) extends ExternalResource {
 
   def getCsvIterator(url: URL, fieldTerminator: Option[String] = None): Iterator[Array[String]] = {
     val inputStream = openStream(url)
-    val reader = Readables.wrap(new InputStreamReader(inputStream, "UTF-8"))
+    val reader = Readables.wrap(new InputStreamReader(inputStream, "UTF-8")
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResources.scala]: Formatting is wrong, in scala we have the `{` on the same line as god intended…

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResources.scala [hunk 2]
```diff
@@ -39,7 +39,10 @@ class CSVResources(cleaner: TaskCloser) extends ExternalResource {
...
+    {
+      override def toString = {url.toString}
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResources.scala]: the `{}` is not needed here, thus 

```
override def toString = url.toString
```

File: community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResourcesTest.scala [hunk 3]
```diff
@@ -175,4 +175,24 @@ class CSVResourcesTest extends CypherFunSuite with CreateTempFileTestSupport {
...
+    val url = createCSVTempFileURL {
+      writer =>
+        // an illegal value. There's currently no other way to verify source description
+        // from a CSV resource (since it returns an iterator) than via an exception
+        // that provides it.
+        writer.println("\"quoted\" and then some")
+    }
+
+    // when
+    try {
```
Comment (systay) [inline community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResourcesTest.scala]: In Scala-land, we usually test exception like this:

```
val e = intercept[IllegalStateException](resources.getCsvIterator(new URL(url)))
e.getMessage() should include(url)
```

**Comment by tinwelint** [comment]
I'll have a look at all your comments tomorrow

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResources.scala (hunk 1 above)
Comment (tinwelint) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/CSVResources.scala]: God also states `And don't be afraid of dedicating one line to a curly bracket, for I will be with you and will protect you`

**Pull request merged**